### PR TITLE
Inherit aws_instance inputs and outputs

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Utilize all `aws_instance` attributes and arguments

## Why
* We want to be compatible with TF resources

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -47,6 +47,16 @@ resource "aws_iam_role" "default" {
   assume_role_policy = "${data.aws_iam_policy_document.default.json}"
 }
 
+locals {
+  iam_role_policy_attachment_count = "${signum(length(var.role_arn)) == 1 ? length(var.role_arn) : 0}"
+}
+
+resource "aws_iam_role_policy_attachment" "default" {
+  count      = "${local.iam_role_policy_attachment_count}"
```
Comment (comeanother) [inline main.tf]: Using `count = length(var.role_arn)`  may be simpler and shorter.

File: main.tf [hunk 2]
```diff
@@ -47,6 +47,16 @@ resource "aws_iam_role" "default" {
...
+  role       = "${aws_iam_role.default.name}"
+  policy_arn = "${element(var.role_arn, count.index)}"
```
Comment (comeanother) [inline main.tf]: Don't understand. Are we trying to attach `role` to a `role`?

File: variables.tf [hunk 3]
```diff
@@ -128,3 +128,8 @@ variable "instance_enabled" {
   description = "Flag for creating an instance. Set to false if it is necessary to skip instance creation"
   default     = true
 }
+
+variable "role_arn" {
```
Comment (comeanother) [inline variables.tf]: Add `description` to a variable.

**Comment by SweetOps** [comment]
has been tested 

```
module "admin_tier" {
  source                      = "../terraform-aws-ec2-instance"
  ssh_key_pair                = "sweetops"
  github_api_token            = ""
  github_organization         = ""
  github_team                 = ""
  vpc_id                      = "some_id"
  security_groups             = ["some_id"]
  subnet                      = "some_id"
  associate_public_ip_address = "true"
  name                        = "kafka"
  namespace                   = "cp"
  stage                       = "dev"
  additional_ips_count        = "1"
  ebs_volume_count            = "2"
}
```

File: security_group.tf [hunk 4]
```diff
@@ -0,0 +1,21 @@
+resource "aws_security_group" "default" {
+  count       = "${local.security_group_count}"
+  name        = "${module.label.id}"
+  vpc_id      = "${var.vpc_id}"
+  description = "Instance default security group (only egress access is allowed)"
+  tags        = "${module.label.tags}"
+
+  egress {
```
Comment (comeanother) [inline security_group.tf]: Use `aws_security_group_rule`.

**Comment by SweetOps** [comment]
screenshot 2017-11-16 13 21 49

File: cloud_watch_alarm.tf [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  dimensions {
+    InstanceId = "${aws_instance.default.id}"
+  }
+
+  alarm_actions = [
+    "${null_resource.check_alarm_action.triggers.action}",
+  ]
+}
+
+resource "null_resource" "eip" {
```
Comment (comeanother) [inline cloud_watch_alarm.tf]: This resource should be moved to `main.tf`.
It doesn't belong to CloudWatch assets.

File: outputs.tf [hunk 6]
```diff
@@ -1,39 +1,54 @@
 output "public_ip" {
```
Comment (comeanother) [inline outputs.tf]: Add `network_interface_id` output.

**Comment by comeanother** [comment]
Can confirm that `user_data` works properly, instance is available.
image

**Comment by comeanother** [comment]
EBS attachment works
image

**Comment by comeanother** [comment]
Additional EIPs work
image

But interface configuration should be added manually

**Pull request merged**